lib/iris/common/resolve.py

```diff
@@ -775,6 +956,20 @@ def _metadata_resolve(self):
         self.lhs_cube_category = self._categorise_items(self.lhs_cube)
         self.rhs_cube_category = self._categorise_items(self.rhs_cube)
 
+        # Categorised dim, aux and scalar coordinate items local to LHS cube only.
+        self.lhs_cube_category_local = _CategoryItems(
+            items_dim=[], items_aux=[], items_scalar=[]
+        )
+        # Categorised dim, aux and scalar coordinate items local to RHS cube only.
+        self.rhs_cube_category_local = _CategoryItems(
+            items_dim=[], items_aux=[], items_scalar=[]
+        )
+        # Categorised dim, aux and scalar coordinate items common to both
+        # LHS cube and RHS cube.
+        self.category_common = _CategoryItems(
+            items_dim=[], items_aux=[], items_scalar=[]
+        )
+
```

Comment:
Do this here, rather in the now defunct `_init`.

And we don't want to do this within `__init__`, as we want it to happen per `__call__`